### PR TITLE
mockgcp: improve fidelity of RedisCluster

### PR DESCRIPTION
**Pull request by justinsb**
- **mockgcp: improve fidelity of RedisCluster**

### CODE REVIEW

File: mockgcp/mockredis/cluster.go [hunk 1]
```diff
@@ -53,7 +53,9 @@ func (r *clusterServer) GetCluster(ctx context.Context, req *pb.GetClusterReques
 		return nil, err
 	}
 
-	return obj, nil
+	retObj := proto.Clone(obj).(*pb.Cluster)
+	retObj.PscConfigs = nil
```
Comment (yuwenma) [inline mockgcp/mockredis/cluster.go:57]: Mind sharing more context about this change?
BTW, `PscConfigs` is immutable by Redis service (which makes the single-field update on a list object becomes easier) .

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve
/hold one question. no blocker

File: mockgcp/mockredis/cluster.go (hunk 1 above)
Comment (justinsb) [reply mockgcp/mockredis/cluster.go:57]: I'll add a comment.  pscConfigs is just not included in these response.

Comment (justinsb) [reply mockgcp/mockredis/cluster.go:58]: Added a comment!

**Comment by justinsb** [comment]
/hold cancel

Added comment

Will still need LGTM (after rebasing / flake)

**Pull request merged**